app/assets/javascripts/discourse/app/controllers/user-topics-list.js

```diff
@@ -54,6 +71,28 @@ export default class UserTopicsListController extends Controller {
     this.pmTopicTrackingState.stopIncomingTracking();
   }
 
+  @action
+  changeSort(sortBy) {
+    if (sortBy === this.resolvedOrder) {
+      this.ascending = !this.resolvedAscending;
+    } else {
+      this.ascending = false;
+    }
+    this.order = sortBy;
+  }
+
+  get resolvedAscending() {
+    if (this.ascending === undefined || this.ascending === null) {
```

Comment:
You can use `import { isNone } from '@ember/utils';` for these type of checks to simplify, e.g. `isNone(this.ascending)`